package.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="2">
+  <name>gz-cmake3</name>
+  <version>3.5.0</version>
```

Comment:
I think we should have a plan for how to keep this version tag in sync with the version in the CMakeLists.txt.

Some options:

1. Manually update version numbers in both places and add a CI check to verify consistency (presumably with a shared github action that we will use in other packages)
2. Make this package.xml version string the single source of truth and add cmake logic to parse the package.xml file to get it (presumably using the existing code that ROS / ament is already using)

I won't block this pull request on that change, but I'd like to at least have an issue open where we continue this conversation

Reply:
I think option 1 would be easier to implement

Reply:
> add cmake logic to parse the package.xml file to get it (presumably using the existing code that ROS / ament is already using)

I believe the following `ament_cmake` macro is used to invoke the `ament_package` python scripts:

* https://github.com/ament/ament_cmake/blob/9743c9cd3e5aef9566acefea6e6e3724e0e5657e/ament_cmake_core/cmake/core/ament_package_xml.cmake
* https://github.com/ament/ament_package

Reply:
I just hacked up a crude github workflow check for consistent versions in the CMakeLists.txt and package.xml file in branch https://github.com/gazebosim/gz-cmake/compare/package_xml...package_xml_ci_test

@j-rivero where would you like to see this CI configuration stored?

Reply:
I should have mentioned I've added a workflow in some of the other package.xml PRs. See https://github.com/gazebosim/gz-sim/actions/runs/8334724361

Reply:
I've added an Action in https://github.com/gazebo-tooling/action-gz-ci/pull/70 and updated this PR to use it.